src/main/java/bio/terra/service/rawls/RawlsService.java

```diff
@@ -26,21 +29,44 @@ public RawlsService(TerraConfiguration terraConfiguration, RawlsClient rawlsClie
     this.terraConfiguration = terraConfiguration;
   }
 
-  public List<WorkspacePolicyModel> resolvePolicyEmails(
+  public record WorkspacePolicyModels(
+      List<WorkspacePolicyModel> accessible, List<InaccessibleWorkspacePolicyModel> inaccessible) {}
+
+  /**
+   * @param samPolicyModel a policy from SAM with member policies
+   * @param userRequest authenticated user
+   * @return workspaces derived from policy emails, both accessible and inaccessible to the user
+   */
+  public WorkspacePolicyModels resolvePolicyEmails(
       SamPolicyModel samPolicyModel, AuthenticatedUserRequest userRequest) {
     Map<UUID, List<ResourcePolicyModel>> workspaceToPolicy =
         samPolicyModel.getMemberPolicies().stream()
             .filter(p -> p.getResourceTypeName().equals(IamResourceType.WORKSPACE.toString()))
             .collect(Collectors.groupingBy(ResourcePolicyModel::getResourceId));
 
-    return workspaceToPolicy.entrySet().stream()
-        .map(
+    List<WorkspacePolicyModel> accessible = new ArrayList<>();
+    List<InaccessibleWorkspacePolicyModel> inaccessible = new ArrayList<>();
+    workspaceToPolicy.entrySet().stream()
+        .forEach(
             entry -> {
               UUID workspaceId = entry.getKey();
-              WorkspaceResponse workspace = rawlsClient.getWorkspace(workspaceId, userRequest);
-              return workspace.toWorkspacePolicyModel(
-                  entry.getValue(), terraConfiguration.getBasePath());
-            })
-        .collect(Collectors.toList());
+              List<ResourcePolicyModel> policies = entry.getValue();
+              try {
+                WorkspaceResponse workspace = rawlsClient.getWorkspace(workspaceId, userRequest);
+                accessible.add(
+                    workspace.toWorkspacePolicyModel(policies, terraConfiguration.getBasePath()));
+              } catch (Exception ex) {
+                inaccessible.add(toInaccessibleWorkspacePolicyModel(workspaceId, policies, ex));
+              }
+            });
+    return new WorkspacePolicyModels(accessible, inaccessible);
```

Comment:
I elected to fork the stream and group workspaces by their accessibility in the response.